pkg/sql/plan2/function/builtin/unary/reverse.go

```diff
@@ -15,31 +15,52 @@
 package unary
 
 import (
+	"errors"
 	"github.com/matrixorigin/matrixone/pkg/container/nulls"
 	"github.com/matrixorigin/matrixone/pkg/container/types"
 	"github.com/matrixorigin/matrixone/pkg/container/vector"
 	"github.com/matrixorigin/matrixone/pkg/vectorize/reverse"
 	"github.com/matrixorigin/matrixone/pkg/vm/process"
 )
 
+var (
+	errorReverseStringFailed = errors.New("errors happened in reversing string")
+)
+
 func Reverse(vectors []*vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	if len(vectors) == 0 || proc == nil {
+		return nil, errorParameterIsInvalid
+	}
+	if vectors[0] == nil {
+		return nil, errorParameterIsInvalid
+	}
 	inputVector := vectors[0]
 	resultType := types.Type{Oid: types.T_varchar, Size: 24}
 	if inputVector.IsScalar() {
 		if inputVector.ConstVectorIsNull() {
 			return proc.AllocScalarNullVector(resultType), nil
 		}
-		inputValues := inputVector.Col.(*types.Bytes)
+		inputValues, ok := inputVector.Col.(*types.Bytes)
+		if !ok {
```

Comment:
I don't think we should add this redundant check here just to pass unit tests, 

during real function execution, this kind of check will be done by TypeCheckFn to guarantee the input vector types, why do we write test inputs that is impossible to happen in matrixone